src/ixdat/techniques/ms.py

```diff
@@ -677,6 +677,132 @@ def gas_flux_calibration(
+                capillary is the same as the flux of pure carrier gas. Defaults
+                to None.
+            mol_conc_ppm (float, list): Concentration of the dilute analyte in
+                the carrier gas in ppm. Defaults to None. Accepts float (for pressure
+                calibration) or list for concentration calibration. If list needs
+                to be same length as tspan_list or selector_list.
+            p_inlet (float, list): Pressure at the inlet (Pa). Overwrites the pressure
+                inherent to self (i.e. the MSInlet object). Accepts float (for conc.
+                calibration) or list for pressure calibration. If list, then
+                needs to be same length as tspan_list or selector_list.
+            t_steady_pulse (float): Length of steady electrolysis for each segment
+                given by selector_list. Defaults to None = entire length of segment
+            tspan_bg (tspan): The time to use as a background
+            ax (Axis): The axis on which to plot the ms_calibration curve result.
+                Defaults to a new axis.
+            axes_measurement (list of Axes): The EC-MS plot axes to highlight the
+                ms_calibration on. Defaults to None. These axes are not returned.
+            return_ax (bool): Whether to return the axis on which the calibration is
+                plotted together with the MSCalResult. Defaults to False.
+
+        Return MSCalResult(, Axis): The result of the ms_calibration (and calibration
+            curve axis if requested) based on flux calculation during selected time
+            periods.
+        TODO: automatically recognize the pressure from measurement (if available)
+        """
+        if not tspan_list:
+            tspan_list = self._get_tspan_list(
+                selector_list, selector_name, t_steady_pulse
+            )
+        if type(mol_conc_ppm) is not list:
+            mol_conc_ppm_list = [mol_conc_ppm for x in tspan_list]
+        else:
+            mol_conc_ppm_list = mol_conc_ppm
+        if type(p_inlet) is not list:
+            p_list = [p_inlet for x in tspan_list]
+        if not len(mol_conc_ppm_list) == len(p_list) == len(tspan_list):
+            raise QuantificationError(
+                "Length of input lists for concentrations"
+                " and tspan or pressures and tspan is not equal"
+            )
+        S_list = []
+        n_dot_list = []
+        for tspan, mol_conc_ppm, pressure in zip(tspan_list, mol_conc_ppm_list, p_list):
+            t, S = measurement.grab_signal(mass, tspan=tspan, tspan_bg=tspan_bg)
+            if axes_measurement:
+                axes_measurement.plot(t, S, color=STANDARD_COLORS[mass], linewidth=5)
+            if carrier_mol:
+                if mol_conc_ppm is not None:
+                    cal_type = "carrier_gas_flux_calibration_curve"
+                else:
+                    raise QuantificationError(
```

Comment:
Can we do this checking of inputs before entering the loop, since I think it needs only be done once? 
Also, I think this error message can be confusing if the user actually meant pure carrier_mol. Can you add something like "For a pure gas, use 'mol' instead of 'carrier_mol' and don't give a 'mol_conc_ppm'" to the error message here, and also below?

Reply:
oh, hadn't thought a step ahead. Checking if `None` is in the list also looks funky. But it makes sense so I think it's fine!